tools/testrunner/log_test.go

```diff
@@ -21,3 +21,40 @@ func TestParseTestTimeouts(t *testing.T) {
 	}, tests)
 	require.Equal(t, string(logOutput), stacktrace)
 }
+
+func TestParseAlerts_DataRaceAndPanic(t *testing.T) {
+	input, err := os.ReadFile("testdata/alerts-input.log")
+	require.NoError(t, err)
+
+	alerts := parseAlerts(string(input))
+	require.NotEmpty(t, alerts)
+
+	// Expect at least one data race and one panic alert.
+	var hasRace, hasPanic bool
+	for _, a := range alerts {
```

Comment:
Should the test be more specific in checking there are 3 alerts (right?) and assert them directly. The loop-based approach makes it a bit indirect.

Reply:
Caught a bug here, there should only be two. The data race wasn't being parsed correctly.